src/RelationViewer/RelationViewerProvider.ts

```diff
@@ -0,0 +1,143 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+
+import {getRelationData} from './example/RelationExample';
+import {RelationViewer} from './RelationViewer';
+
+export class RelationViewerDocument implements vscode.CustomDocument {
+  private readonly _uri: vscode.Uri;
+  private _relationViewer: RelationViewer[];
+
+  static async create(uri: vscode.Uri):
+      Promise<RelationViewerDocument|PromiseLike<RelationViewerDocument>> {
+    return new RelationViewerDocument(uri);
+  }
+
+  private constructor(uri: vscode.Uri) {
+    this._uri = uri;
+    this._relationViewer = [];
+  }
+
+  public get uri() {
+    return this._uri;
+  }
+
+  // CustomDocument implements
+  dispose(): void {
+    // NOTE panel is closed before document and this is just for safety
+    this._relationViewer.forEach((view) => {
+      while (this._relationViewer.length) {
+        view.disposeMetadataView();
+      }
+    });
+    this._relationViewer = [];
+  }
+
+  public openView(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, fileUri: vscode.Uri) {
+    let view = new RelationViewer(panel, extensionUri);
+
+    view.initWebview();
+    view.loadContent();
+    this._relationViewer.push(view);
+
+    const payload = getRelationData(fileUri);
+
+    // Send a message the relation data to the web view
+    panel.webview.postMessage({type: 'create', payload: payload});
+
+    panel.onDidDispose(() => {
+      // TODO make faster
+      this._relationViewer.forEach((view, index) => {
+        if (view.owner(panel)) {
+          view.disposeMetadataView();
+          this._relationViewer.splice(index, 1);
+        }
+      });
+    });
+
+    return view;
+  }
+}
+
+export class RelationViewerProvider implements
+    vscode.CustomReadonlyEditorProvider<RelationViewerDocument> {
+  public static readonly viewType = 'one.viewer.relation';
+  private _context: vscode.ExtensionContext;
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new RelationViewerProvider(context);
+
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(RelationViewerProvider.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        }
+      }),
+      vscode.commands.registerCommand(
+          'one.viewer.relation.showFromDefaultExplorer',
+          async (uri) => {
+            const fileUri = uri;
+
+            vscode.commands.executeCommand(
+                'vscode.openWith', fileUri, RelationViewerProvider.viewType);
+          }),
+      vscode.commands.registerCommand(
+          'one.viewer.relation.showFromOneExplorer',
+          async (uri) => {
+            // If the method is executed in the ONE Explorer, change the uri instance.
+            const fileUri = uri.uri;
```

Comment:
```suggestion
            const fileUri = node.uri;
```

Because, actually a 'Node' type is coming in.

Reply:
ok, I`ll apply it!